File: backend/api/src/main/java/com/github/ai/leetcodequiz/api/ErrorMessageDto.java
package com.github.ai.leetcodequiz.api;

import org.jetbrains.annotations.NotNull;

import java.util.List;
import java.util.Optional;

public record ErrorMessageDto(
    @NotNull String message,
    @NotNull String exception,
    @NotNull String stacktraceBase64,
    @NotNull List<String> stacktraceLines
) {
}

File: backend/api/src/main/scala/com/github/ai/leetcodequiz/api/ErrorMessageDto.scala
package com.github.ai.leetcodequiz.api

import zio.json.{JsonDecoder, JsonEncoder}

case class ErrorMessageDto(
  message: String,
  exception: String,
  stacktraceBase64: String,
  stacktraceLines: List[String]
) derives JsonEncoder,
      JsonDecoder
